src/tests/ftest/pool/Attribute.py

```diff
@@ -0,0 +1,242 @@
+                self.large_data_set = {}
+
+        except Exception as e:
+            print ("In the setup exception handler\n")
+            print (e)
+            print (traceback.format_exc())
+
+    def tearDown(self):
+        try:
+            if self.pool is not None:
+                self.pool.disconnect()
+                self.pool.destroy(1)
+        finally:
+            ServerUtils.stopServer(hosts=self.hostlist)
+
+    def create_data_set(self):
+        """
+        To create the large attribute dictionary
+        """
+        allchar = string.ascii_letters + string.digits
+        for i in range(1024):
+            self.large_data_set[str(i)] = "".join(random.choice(allchar)
+                 for x in range(random.randint(1, 100)))
+
+    def test_pool_attributes(self):
+        """
+        Test ID: DAOS-1359
+
+        Test description: Test basic pool attribute tests (sync).
+
+        :avocado: tags=regression,pool,pool_attr,attribute,sync_poolattribute
+        """
+        expected_for_param = []
+        name = self.params.get("name", '/run/attrtests/name_handles/*/')
+        expected_for_param.append(name[1])
+        value = self.params.get("value", '/run/attrtests/value_handles/*/')
+        expected_for_param.append(value[1])
+
+        attr_dict = {name[0]:value[0]}
+        if name[0] is not None:
+            if "largenumberofattr" in name[0]:
+                self.create_data_set()
+                attr_dict = self.large_data_set
+
+        expected_result = 'PASS'
+        for result in expected_for_param:
+            if result == 'FAIL':
+                expected_result = 'FAIL'
+                break
+        try:
+            self.pool.set_attr(data=attr_dict)
+            size, buf = self.pool.list_attr()
+
+            verify_list_attr(attr_dict, size.value, buf, "sync")
+
+            if name[0] is not None:
+                # Request something that doesn't exist
+                if "Negative" in name[0]:
+                    name[0] = "rubbish"
+                # large attr test messes with the dictionary so skip
+                # the get test
+                if "largenumberofattr" not in name[0]:
+                    results = {}
+                    results = self.pool.get_attr([name[0]])
+                    verify_get_attr(attr_dict, results)
+
+            if expected_result in ['FAIL']:
+                self.fail("Test was expected to fail but it passed.\n")
+            pass
+
+        except DaosApiError as e:
+            print (e)
+            print (traceback.format_exc())
+            if expected_result == 'PASS':
+                self.fail("Test was expected to pass but it failed.\n")
+
+    def test_pool_attribute_asyn(self):
+        """
+        Test ID: DAOS-1359
+
+        Test description: Test basic pool attribute tests (async).
+
+        :avocado: tags=regression,pool,pool_attr,attribute,async_poolattribute
+        """
+        global GLOB_SIGNAL
+        global GLOB_RC
```

Comment:
(pylint-global-variable-not-assigned) Using global for 'GLOB_RC' but no assignment is done